src/Mailgun/Connection/RestClient.php

```diff
@@ -147,6 +147,9 @@ public function post($endpointUrl, array $postData = [], $files = [])
                     $postDataMultipart[] = [
                         'name' => sprintf('%s[%d]', $key, $index++),
                         'contents' => $subValue,
+                        'headers' => [
+                            'Content-Disposition' => sprintf('form-data; name="%s"', $key),
```

Comment:
Well, the Mailgun server kind of understands it (it receives all arguments but they are disordered). I think it's best if we wait for Mailgun (either by responding to my support ticket or here) to clear things up. All the other clients/libraries I checked (curl, Go, ...) do not use the array syntax.